src/simulator_processing/infrared_id_compensation.cpp

```diff
@@ -54,14 +54,19 @@ bool InfraredIdCompensation::setupFromRos(const ros::NodeHandle& nh,
 }
 
 void InfraredIdCompensation::imageCallback(const sensor_msgs::ImagePtr& msg) {
-  cv_bridge::CvImagePtr img = cv_bridge::toCvCopy(msg, msg->encoding);
-  for (int v = 0; v < img->image.rows; v++) {
-    for (int u = 0; u < img->image.cols; u++) {
-      img->image.at<uchar>(v, u) =
-          infrared_compensation_[img->image.at<uchar>(v, u)];
+  try {
+    cv_bridge::CvImagePtr img = cv_bridge::toCvCopy(msg, msg->encoding);
+    for (int v = 0; v < img->image.rows; v++) {
+      for (int u = 0; u < img->image.cols; u++) {
+        img->image.at<uchar>(v, u) =
+            infrared_compensation_[img->image.at<uchar>(v, u)];
+      }
     }
+    pub_.publish(img->toImageMsg());
+  } catch (const cv_bridge::Exception& e) {
+    std::cerr << "Could not convert image to CvImage\n";
```

Comment:
Better to stay consistent with logging (i.e. we use glog throughout the project). Consider replacing with more informative text, e.g. LOG(WARNING) << "Could not convert image for infrared_id_compensation, skipping frame (" << e.what() << ").";